### PR TITLE
v1.6.7 restrictions endpoints [BAC-2336] [BAC-2365]

### PR DESCRIPTION
**Pull request by jiajames**
See [v3.md](https://github.com/dydxprotocol/proposals/blob/master/proposals/api/v3.MD#get-v3restrictions)

tested most compliance flows on staging

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -160,6 +160,15 @@ export enum NftRevealType {
   WEEK = 'WEEK',
 }
 
+export enum AddressRestrictionType {
```
Comment (Christopher-Li) [inline src/types.ts:163]: Shouldn't this just be named `RestrictionType`? Why should it be narrowed down to Address?

File: src/modules/private.ts [hunk 2]
```diff
@@ -883,6 +885,43 @@ export default class Private {
...
+      {
+        ...genericParams,
+      },
+    );
+  }
+
+  /**
+   * @description comply to dYdX terms of service after a first offense.
+   */
+  async userCompliance(
```
Comment (Christopher-Li) [inline src/modules/private.ts:905]: this is for external use right? Shouldn't we name this something that reflects the external api? Like `postRestrictionsCompliance`?

Comment (jiajames) [reply src/modules/private.ts]: done

File: src/types.ts (hunk 1 above)
Comment (jiajames) [reply src/types.ts:163]: all restrictions are based on address, so wanted to make that clear here

**Review by Christopher-Li — APPROVED** [review]

**Pull request merged**